### PR TITLE
TEACH-1415 Units not appearing on course edit page

### PR DESCRIPTION
**Pull request by vijayamanohararaj**
With the change (https://github.com/code-dot-org/code-dot-org/commit/04c7484c1c56b8894ea5dac8923c8db74820da59) to limit every unit can be part of only one unit group, the list that populates the drop down in the course edit page was filtered only to units that are not part of any unit groups. As a result, the dropdowns did not populate values that are part of the current unit as well. 

This change 
1. Fixes that by additionally adding the current unit in the list of options in the client side. 
2. Adds the new unit test for this change using RTL, which shows up as lot more changes than there actually are in the diff. 

## Links

JIRA: https://codedotorg.atlassian.net/browse/TEACH-1415

## Testing story

- Validated manually locally and new UTs
- Drone

## Deployment strategy

Regular DTP

## Follow-up work

None

## Privacy

N/A

## Security

N/A

## Caching

N/A

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/levelbuilder/course-editor/CourseEditorTest.js [hunk 1]
```diff
@@ -63,6 +36,33 @@ describe('CourseEditor', () => {
       resources: createResourcesReducer('teacherResource'),
       studentResources: createResourcesReducer('studentResource'),
     });
+    store = getStore();
+
+    defaultProps = {
```
Comment (davidsbailey) [inline apps/test/unit/levelbuilder/course-editor/CourseEditorTest.js]: please use const instead of let, since the value does not change

File: apps/src/levelbuilder/course-editor/CourseEditor.js [hunk 2]
```diff
@@ -416,7 +416,8 @@ class CourseEditor extends Component {
             <div>
               The dropdown(s) below represent the ordered set of units in this
               course. To remove a unit, just set the dropdown to the default
-              (first) value.
+              (first) value. Please referesh the page after every edit to get
+              the most uptodate list of units.
```
Comment (davidsbailey) [inline apps/src/levelbuilder/course-editor/CourseEditor.js]: why is this necessary?

Comment (vijayamanohararaj) [reply apps/src/levelbuilder/course-editor/CourseEditor.js]: I was mainly thinking of a scenario where we want to reorder units. In that case, one of the units that is currently in the unit group would be removed, but this won't be available in the other drop downs. I'm thinking there can be a way to address this by adding all the units in current course/unit group to the list of options. I'll explore this a bit more.

Comment (davidsbailey) [reply apps/src/levelbuilder/course-editor/CourseEditor.js]: > I'm thinking there can be a way to address this by adding all the units in current course/unit group to the list of options.

Yes, this sounds promising!

The other test case which comes to mind is, if I have a list of unit in a course:
csp1
csp2
csp3

what happens if I select csp1 and change it to csp3. I think you would want it to end up like this:
csp3
csp2
[blank]

Comment (vijayamanohararaj) [reply apps/src/levelbuilder/course-editor/CourseEditor.js]: Yeah, I was thinking of that as well. I am hoping there is some backend validation that will block have the same unit twice in a unit group. I'd guess the client side behavior today would be that the lists will show up as

csp3
csp2
csp3

and I'm thinking as long as backend blocks it, this should be still ok. I'll test this locally to confirm.

Comment (davidsbailey) [reply apps/src/levelbuilder/course-editor/CourseEditor.js]: I agree, it could be ok for the client to be in a "bad" state temporarily as long as we have validations on the backend to keep us from getting into a bad state.

Comment (vijayamanohararaj) [reply apps/src/levelbuilder/course-editor/CourseEditor.js]: I tested these two scenarios with the latest changes

1. Reordering units - this works as expected
2. Add the same unit twice in the course - in this case the client side doesn't throw any error, but in the backend, only one unit is getting saved. Captured this in a video 


https://github.com/user-attachments/assets/44b3e15e-b422-4ff6-93c3-630def4e4e00

Comment (davidsbailey) [reply apps/src/levelbuilder/course-editor/CourseEditor.js]: I think this experience is good enough. it would be ideal for us to show an red error message here instead of green success, but reordering units is uncommon today (and very unlikely for an error to go unnoticed) so this seems fine for now. it may become more common when we tackle modularity next quarter, but we can update this logic at that time to deal with that scenario.

**Review by davidsbailey — APPROVED** [review]
